dhall-openapi/src/Dhall/Kubernetes/Convert.hs

```diff
@@ -315,46 +315,27 @@ toDefault prefixMap definitions modelName = go
       = Dhall.Embed $ mkImport prefixMap ["types", ".."] (file <> ".dhall")
     adjustImport other = other
 
-
--- | Get a Dhall.Map filled with imports, for creating giant Records or Unions of types or defaults
-getImportsMap
-  :: Data.Map.Map Prefix Dhall.Import -- ^ Mapping of prefixes to import roots
-  -> DuplicateHandler                 -- ^ Duplicate name handler
-  -> [ModelName]                      -- ^ A list of all the object names
-  -> Text                             -- ^ The folder we should get imports from
-  -> [ModelName]                      -- ^ List of the object names we want to include in the Map
-  -> Dhall.Map.Map Text Expr
-getImportsMap prefixMap duplicateNameHandler objectNames folder toInclude
-  = Dhall.Map.fromList
-  $ Data.Map.elems
-  -- This intersection is here to "pick" common elements between "all the objects"
-  -- and "objects we want to include", already associating keys to their import
-  $ Data.Map.intersectionWithKey
-      (\(ModelName name) key _ -> (key, Dhall.Embed $ mkImport prefixMap [folder] (name <> ".dhall")))
-      namespacedToSimple
-      (Data.Map.fromList $ fmap (,()) toInclude)
+-- | Given a list of fully namespaced models, it will group them by the
+--   object name resolving duplicates with the given function
+groupBySimpleModelName :: DuplicateHandler -> [ModelName] -> Data.Map.Map SimpleModelName ModelName
+groupBySimpleModelName duplicateNameHandler models = Data.Map.mapMaybe selectObject $ groupBySimpleName models
   where
-    -- | A map from namespaced names to simple ones (i.e. without the namespace)
-    namespacedToSimple
-      = Data.Map.fromList $ mapMaybe selectObject $ Data.Map.toList $ groupByObjectName objectNames
-
-    -- | Given a list of fully namespaced objects, it will group them by the
-    --   object name
-    groupByObjectName :: [ModelName] -> Data.Map.Map Text [ModelName]
-    groupByObjectName modelNames = Data.Map.unionsWith (<>)
-      $ (\name -> Data.Map.singleton (getKind name) [name])
+    groupBySimpleName :: [ModelName] -> Data.Map.Map SimpleModelName [ModelName]
+    groupBySimpleName modelNames = Data.Map.unionsWith (<>)
+      $ (\name -> Data.Map.singleton (toSimpleModelName name) [name])
       <$> modelNames
-      where
-        getKind (ModelName name) =
-          let elems = Text.split (== '.') name
-          in elems List.!! (length elems - 1)
+
+    toSimpleModelName :: ModelName -> SimpleModelName
+    toSimpleModelName (ModelName name) =
+      let elems = Text.split (== '.') name
+      in elems List.!! (length elems - 1)
```

Comment:
This can be made safer by defining a wrapper around `Text.split` that returns a `NonEmpty Text`:

```haskell
split :: (Char -> Bool) -> Text -> NonEmpty Text
split predicate text =
    case Text.split predicate text of
        []     -> "" :| []
        x : xs ->  x :| xs
```

… which is really what the type of `Text.split` should be anyway (since it never returns an empty list).  Then you can simplify this to:

```haskell
toSimpleModelName (ModelName name) = NonEmpty.last (split (== '.') name)
```